src/pages/recruitment/recruitment.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { Fragment, useState } from 'react';
+import styled from 'styled-components';
+import config from './formConfig';
+
+const FormWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  width: 50%;
+`;
+
+const InputWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  margin: 20px 0;
+`;
+
+const Input = styled.input`
+  margin-left: 10px;
+  outline: none;
+
+  border-color: ${(props) => props.inValid && 'red'}
+`;
+
+const ErrorMessage = styled.div`
+  color: red;
+  display: flex;
+  justify-content: flex-end;
+`;
+
+const SubmitButton = styled.button`
+  width: 100%;
+
+  &:hover {
+    cursor: pointer;
+  }
+`;
+
+export default function RecruitmentPage() {
+  const [form, updateForm] = useState(
+    Object.keys(config).reduce(
+      (formObject, key) => ({
+        ...formObject,
+        [key]: '',
+      }),
+      {},
+    ),
+  );
+
+  const [touched, setTouched] = useState(false);
+
+  const formKeys = Object.keys(form);
+
+  const formValues = formKeys.reduce(
+    (formValuesObject, key) => ({
+      ...formValuesObject,
+      [key]: form[key],
+    }),
+    {},
+  );
+
+  const findEmptyField = () => {
+    const emptyField = formKeys.find((key) => !formValues[key]);
+
+    return !!emptyField;
+  };
+
+  const validateInput = () => {
+    const inValidInputKey = formKeys.find((key) => (
+      config[key].isInValid?.(formValues[key], formValues)
+    ));
+
+    return !inValidInputKey;
+  };
+
+  const inputHandler = (formKey) => (
+    (event) => {
+      event.preventDefault();
+
+      updateForm((prevState) => ({
+        ...prevState,
+        [formKey]: event.target.value,
+      }));
+    }
+  );
+
+  const submitHandler = (event) => {
+    event.preventDefault();
+
+    if (findEmptyField() || !validateInput()) {
+      setTouched(true);
+    }
+  };
+
+  return (
+    <FormWrapper>
+      <h1>
+        成为团长
+      </h1>
+
+      <form onSubmit={submitHandler}>
+        {Object.keys(config).map((key) => (
+          <Fragment key={key}>
+            <InputWrapper>
+              <label htmlFor={key}>
+                {config[key].label}
+              </label>
+              <Input
+                type={config[key].inputType || 'text'}
+                name={key}
+                onChange={inputHandler(key)}
+                value={formValues[key]}
+                inValid={touched && (
```

Comment:
用 error 代表 error 情况下的 input
```suggestion
                error={touched && (
```